runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -331,11 +354,30 @@ public Token getTrimMark() {
     }
 
     /**
-     * Get the last address in the address space
+     * Get the log's tail, i.e., last address in the address space.
+     */
+    public Long getLogTail() {
+        return layoutHelper(
+                e -> Utils.getLogTail(e.getLayout(), runtime));
+    }
+
+    /**
+     * Get all tails, includes: log tail and stream tails.
      */
     public TailsResponse getAllTails() {
         return layoutHelper(
-                e -> getTails(e.getLayout(), runtime));
+                e -> Utils.getAllTails(e.getLayout(), runtime));
```

Comment:
why `getLogTail` in `Utils` class? It's semantically wrong. Please create another class for that

Reply:
I agree but as that was there let's leave this for a separate cleanup, I'll track it as an issue to not bring in another change.